src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOQuerySupport.scala

```diff
@@ -100,6 +99,15 @@ trait ElasticSearchDAOQuerySupport extends ElasticSearchDAOSupport {
       rp => query.must(researchPurposeSupport.researchPurposeFilters(rp, toLibraryAttributeName))
     }
 
+    val groupsQuery = boolQuery
+    // https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-exists-query.html
+    groupsQuery.should(boolQuery.mustNot(existsQuery(fieldDiscoverableByGroups)))
+    if (groups.nonEmpty) {
+      groupsQuery.should(termsQuery(fieldDiscoverableByGroups, groups.asJavaCollection))
+    }
+    groupsQuery.should(termsQuery(fieldWorkspaceId+".keyword" , workspaceIds.asJavaCollection))
```

Comment:
I'm trying to think of a case in which `workspaceIds` will be empty - and I can't come up with it - but might want to protect against that case.

On  the flip side - have you tested this with arbitrarily large numbers of workspaceIds? Is there *any* limit on size? What if the user has tens of thousands of workspaces?

Reply:
i can't remember now. some of the documentation says there is no explicit limit, but performance will degrade. months ago when i first tried out this change in implementation, i did test by adding thousands of ids to the list but i can't remember exactly how many :(